src/Engine/ProtoCore/Utils/CoreUtils.cs

```diff
@@ -736,14 +737,40 @@ public static string GetIdentifierStringUntilFirstParenthesis(ProtoCore.AST.Asso
         }
 
         /// <summary>
+        /// Retrieves the string format of the identifier list from left to right, leaving out any symbols after the last identifier.
+        /// Given: A.B()
+        ///     Return: "A"
+        /// Given: A.B.C()[0]
+        ///     Return: "A.B"
+        /// Given: A.B().C
+        ///     Return: "A"
+        /// Given: A.B[0].C
+        ///     Return: "A.B[0].C"
+        /// </summary>
+        /// <param name="identList"></param>
+        /// <returns></returns>
```

Comment:
The above `summary` is great, except that we do not have any test written for this new method. Please add all the above conditions, and then check their return values. Also, please be sure to add negative test cases also (`null`, empty string, etc).